src/nlp/utils/file_utils.py

```diff
@@ -216,7 +215,7 @@ def cached_path(url_or_filename, download_config=None, **download_kwargs,) -> Op
         # We avoid '.' in dir name and add "-extracted" at the end: "./model.zip" => "./model-zip-extracted/"
         output_dir, output_file = os.path.split(output_path)
         output_extract_dir_name = output_file.replace(".", "-") + "-extracted"
-        output_path_extracted = os.path.join(output_dir, output_extract_dir_name)
+        output_path_extracted = os.path.join(cache_dir, output_extract_dir_name)
```

Comment:
So `output_dir` is not needed anymore? Is that safe in a sense that `output_path_extracted` is unique for every dataset if we don't use the `output_dir`? Maybe safer to do `output_path_extracted = os.path.join(cache_dir, output_dir, output_extract_dir_name)`

And are we replacing the `output_dir` here only because of the tests? We could also just put `dummy_data-extracted` in `.gitignore`...

Reply:
I see. There might be collisions in names. However i like the idea of extracting in the cache directory and not right next to the zipped file (in general, not only for tests).

Reply:
Agree! Do you think we could go with `output_path_extracted = os.path.join(cache_dir, output_dir, output_extract_dir_name)` ?

Maybe @thomwolf can chime in here as well

Reply:
`output_dir` can be inside many subfolders. We could re-use the `url_to_filename` to hash it like a url ? Something like `os.path.join(cache_dir, url_to_filename(output_path))` ?